opentasks-provider/src/main/java/org/dmfs/provider/tasks/utils/Zipped.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.provider.tasks.utils;
+
+import org.dmfs.jems.function.BiFunction;
+import org.dmfs.jems.optional.decorators.Mapped;
+import org.dmfs.jems.single.Single;
+import org.dmfs.jems.single.combined.Backed;
+import org.dmfs.jems.single.decorators.DelegatingSingle;
+import org.dmfs.optional.Optional;
+
+
+/**
+ * Experimental {@link Single} which applies a {@link BiFunction} based on the presence of an {@link Optional}.
```

Comment:
I see (at least) two big advantages for putting this into a separate class:
* it's easier to read and understand if it has a proper name
* it's testable in isolation
We can never be sure if we remember already having special classes for the things we implement. That's one of the reason we do peer reviews, It doubles the chance of someone spotting duplicate implementations. If we don't have a special class we'll always have to implement it again. We we have a special class, chances are that we remember having it.
I mean this is something very generic that could be useful in other scenarios as well and so it could go into jems.

Reply:
Ok, if you see this as something that can come commonly useful, then of course it makes sense to have this, in jems.
I am not sure about the name, but don't we use `Zipped` for things that create `Pair`s?
Maybe using 'fold' or 'reduce'?

Reply:
For the time being I keep the name `Zipped`. We can think of a better one when we move it to jems.